crates/interpreter/src/instructions/i256.rs

```diff
@@ -105,9 +105,11 @@ pub fn i256_mod(mut first: U256, mut second: U256) -> U256 {
         return U256::ZERO;
     }
 
-    let _ = i256_sign_compl(&mut second);
+    let second_sign = i256_sign_compl(&mut second);
+    if second_sign == Sign::Zero {
+        return U256::ZERO;
+    }
```

Comment:
Also, just curious, what was the purpose of this before?

Reply:
I am not sure, probably just an artefact of code moving